src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -177,7 +164,49 @@ int IVsEditorFactory.Close()
             return VSConstants.S_OK;
         }
 
-        private string? GetWinFormsLoaderName(IVsHierarchy vsHierarchy)
+        public object GetDocumentData(uint grfCreate, string pszMkDocument, IVsHierarchy pHier, uint itemid)
+        {
+            var editorAdaptersFactoryService = _componentModel.GetService<IVsEditorAdaptersFactoryService>();
+
+            var contentTypeRegistryService = _componentModel.GetService<IContentTypeRegistryService>();
+            var contentType = contentTypeRegistryService.GetContentType(ContentTypeName);
+            var textBuffer = editorAdaptersFactoryService.CreateVsTextBufferAdapter(_oleServiceProvider, contentType);
+
+            if (_encoding)
+            {
+                if (textBuffer is IVsUserData userData)
+                {
+                    // The editor shims require that the boxed value when setting the PromptOnLoad flag is a uint
+                    var hresult = userData.SetData(
+                        VSConstants.VsTextBufferUserDataGuid.VsBufferEncodingPromptOnLoad_guid,
+                        (uint)__PROMPTONLOADFLAGS.codepagePrompt);
+
+                    Marshal.ThrowExceptionForHR(hresult);
+                }
+            }
+
+            return textBuffer;
+        }
+
+        public object GetDocumentView(uint grfCreate, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, uint itemid)
+        {
+            // There is no scenario need currently to implement this method.
+            throw new NotImplementedException();
+        }
+
+        public string GetEditorCaption(string pszMkDocument, string pszPhysicalView, IVsHierarchy pHier, IntPtr punkDocData, out Guid pguidCmdUI)
+        {
+            // It is not possible to get this information without initializing the designer.
+            // There is no other scenario need currently to implement this method.
+            throw new NotImplementedException();
+        }
+
+        public bool ShouldDeferUntilIntellisenseIsReady(uint grfCreate, string pszMkDocument, string pszPhysicalView)
+        {
+            return "Form".Equals(pszPhysicalView, StringComparison.OrdinalIgnoreCase);
```

Comment:
What happens with Component Designer? Is it also set to "Form"?

Reply:
I believe so. The C# and VB editor factories have only ever checked for "Form" or "Code".